PhysicsTools/TensorFlow/interface/Graph.h

```diff
@@ -0,0 +1,165 @@
+/*
+ * TensorFlow graph interface.
+ * Based on TensorFlow C API 1.1.
+ * For more info, see https://gitlab.cern.ch/mrieger/CMSSW-DNN.
+ *
+ * Author: Marcel Rieger
+ */
+
+#ifndef PHYSICSTOOLS_TENSORFLOW_GRAPH_H
+#define PHYSICSTOOLS_TENSORFLOW_GRAPH_H
+
+#include <algorithm>
+#include <stdexcept>
+#include <string>
+
+#include <tensorflow/c/c_api.h>
+
+#include "PhysicsTools/TensorFlow/interface/Tensor.h"
+
+namespace tf
+{
+
+// generic class containing all information of inputs to / ouptuts from a graph
+class GraphIO
+{
+public:
+    // constructur
+    GraphIO(Tensor* tensor, TF_Operation* tf_operation, const std::string& opName, int opIndex = 0);
+
+    // destructor
+    virtual ~GraphIO();
+
+    // returns the pointer to the tensor instance
+    inline Tensor* getTensor()
+    {
+        return tensor;
+    }
+
+    // returns a reference to the tensorflow output object
+    inline TF_Output& getTFOutput()
+    {
+        return tf_output;
+    }
+
+    // returns the operation name
+    inline std::string& getOpName()
+    {
+        return opName;
+    }
+
+    // returns the operation index
+    inline int getOpIndex()
+    {
+        return opIndex;
+    }
+
+private:
+    Tensor* tensor;
+    std::string opName;
+    int opIndex;
+
+    TF_Output tf_output;
```

Comment:
here and in other classes of PhysicsTools/TensorFlow
please change data member names to have a different style from locals (e.g. add a trailing underscore).